src/panels/ai/ChatPanelSection.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react';
+import { VscMegaphone, VscRunAll, VscWand } from 'react-icons/vsc';
+
+import { chatWithAI } from '../../api/chat';
+import { PanelSection } from '../../components/PanelSection';
+import { cx } from '../../cx';
+
+interface ChatThreadItem {
+  id: string;
+  question: string;
+  answer?: string;
+}
+type ChatThread = ChatThreadItem[];
+
+export function ChatPanelSection() {
+  const [text, setText] = useState('');
+
+  const [chatThread, setChatThread] = useState<ChatThread>([
+    // {
+    //   id: '1',
+    //   question: 'What is your name?',
+    //   answer:
+    //     'I am an AI language model and do not have a name. How may I assist you with your query related to the given context?',
+    // },
+    // {
+    //   id: '2',
+    //   question: 'What can you tell me about hidden feedback loops in machine learning?',
+    //   answer:
+    //     'Hidden feedback loops in machine learning refer to situations where two systems indirectly influence each other through the world, leading to changes in behavior that may not be immediately visible. These loops may exist between completely disjoint systems and can make analyzing the effect of proposed changes extremely difficult, adding cost to even simple improvements. It is recommended to look carefully for hidden feedback loops and remove them whenever feasible.',
+    // },
+  ]);
+  const [currentChatThreadItem, setCurrentChatThreadItem] = useState<ChatThreadItem | null>(null);
```

Comment:
Do you want to move the thread state into atoms? Presumably the interactions will grow more elaborate in the future and we'll want it to persist even if you close the component. Alternatively the thread state could live in the backend.